### PR TITLE
HW Nr.2 Create 2 jobs and e2e test for these jobs

### PR DESCRIPTION
**Pull request by AndriiLiapko**

### CODE REVIEW

**Comment by AndriiLiapko** [comment]
@mikolaz27

File: src/lesson_02/defs.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+from typing import Dict
+
+API_ENDPOINT: str = 'https://fake-api-vycpfa6oca-uc.a.run.app/sales'
+
+SCHEMA: Dict = {
```
Comment (mikolaz27) [inline src/lesson_02/defs.py:5]: settings name looks better

File: src/lesson_02/job1.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import logging
+
+from flask import Flask, request, jsonify, Response
+
+app = Flask(__name__)
+
+app.logger.addHandler(logging.StreamHandler())
+app.logger.setLevel(logging.INFO)
+
+API_ENDPOINT: str = 'https://fake-api-vycpfa6oca-uc.a.run.app/sales'
```
Comment (mikolaz27) [inline src/lesson_02/job1.py:14]: take from settings

File: src/lesson_02/job1.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+
+def fetch_data(endpoint: str, date: str, page: int) -> Response:
+    return requests.get(
```
Comment (mikolaz27) [inline src/lesson_02/job1.py:18]: this part better to move to separate module

File: src/lesson_02/job1.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+        app.logger.info(f'Fetch date={target_date}, page={page} Response code: {response.status_code}')
+
+        content = response.json()
+        file_path = os.path.join(target_path, f"sales_{target_date}_{page}.json")
+        with open(file_path, 'w') as f:
+            json.dump(content, f)
+
+        app.logger.info(f"Saving file to {file_path}")
+
```
Comment (mikolaz27) [inline src/lesson_02/job1.py:56]: same here, split it by functions.

File: src/lesson_02/job1.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    app.logger.info(f'Data parsed: path={target_path}, date={target_date}')
+
+    if os.path.exists(target_path):
+        shutil.rmtree(target_path)
+
+    os.makedirs(target_path)
+
+    page: int = 1
+    while True:
```
Comment (mikolaz27) [inline src/lesson_02/job1.py:41]: you can use condition while response != 404

File: src/lesson_02/job1.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        response = fetch_data(API_ENDPOINT, target_date, page)
+
+        if response.status_code == 404:
```
Comment (mikolaz27) [inline src/lesson_02/job1.py:44]: use name from http lib

File: src/lesson_02/job2.py [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+from flask import Flask, request, jsonify
+
+from lesson_02.defs import SCHEMA
+
+app = Flask(__name__)
+
+app.logger.addHandler(logging.StreamHandler())
+app.logger.setLevel(logging.INFO)
+
+PARSED_SCHEMA = parse_schema(SCHEMA)
```
Comment (mikolaz27) [inline src/lesson_02/job2.py:16]: move to settings

File: src/lesson_02/job1.py (hunk 2 above)
Comment (AndriiLiapko) [reply src/lesson_02/job1.py]: moved

File: src/lesson_02/job1.py (hunk 5 above)
Comment (AndriiLiapko) [reply src/lesson_02/job1.py]: done

File: src/lesson_02/job2.py (hunk 7 above)
Comment (AndriiLiapko) [reply src/lesson_02/job2.py]: moved

File: src/lesson_02/job1.py (hunk 6 above)
Comment (AndriiLiapko) [reply src/lesson_02/job1.py]: refactored

File: src/lesson_02/job1.py (hunk 3 above)
Comment (AndriiLiapko) [reply src/lesson_02/job1.py]: created utils file as a container for common functions

File: src/lesson_02/job1.py (hunk 4 above)
Comment (AndriiLiapko) [reply src/lesson_02/job1.py]: done

**Review by mikolaz27 — APPROVED** [review]

**Pull request merged**